File: server/models/Room.js
const mongoose = require('mongoose');

const roomSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
        trim: true
    },
    description: {
        type: String,
        default: ''
    },
    price: {
        type: Number,
        required: true,
        min: 0
    },
    capacity: {
        type: Number,
        required: true,
        min: 1
    },
    imageUrl: {
        type: String,
        default: ''
    },
    maxSlots: {
        type: Number,
        default: 1,
        min: 1
    },
    location: {
        type: String,
        default: ''
    },
    address: {
        type: String,
        default: ''
    },
    rating: {
        type: Number,
        default: 0,
        min: 0,
        max: 5
    },
    reviewCount: {
        type: Number,
        default: 0,
        min: 0
    },
    amenities: {
        type: String,
        default: ''
    },
    roomType: {
        type: String,
        default: ''
    },
    area: {
        type: Number,
        default: 0,
        min: 0
    },
    createdAt: {
        type: Date,
        default: Date.now
    }
});

module.exports = mongoose.model('Room', roomSchema);

File: app/src/main/java/com/example/homestay/data/entity/Room.kt
package com.example.homestay.data.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "rooms")
data class Room(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val mongoId: String? = null, // MongoDB ObjectId - để sync với backend
    val name: String,
    val description: String,
    val price: Double,
    val imageUrl: String,
    val location: String,
    val address: String,
    val rating: Float = 0f,
    val reviewCount: Int = 0,
    val amenities: String, // JSON string hoặc comma-separated
    val maxGuests: Int = 2,
    val roomType: String, // "Phòng đơn", "Phòng đôi", "Phòng gia đình"
    val area: Int = 0, // diện tích m²
    val maxSlots: Int = 1, // Số slot tối đa của phòng (giới hạn số booking cùng lúc)
    val isAvailable: Boolean = true
)

